src/Orleans/AssemblyLoader/AssemblyLoader.cs

```diff
@@ -319,9 +319,16 @@ private bool ReflectionOnlyLoadAssembly(string pathName, out Assembly assembly,
             {
                 if (SimulateReflectionOnlyLoadFailure)
                     throw NewTestUnexpectedException();
-                
+
                 assembly = Assembly.ReflectionOnlyLoadFrom(pathName);
             }
+            catch (BadImageFormatException)
+            {
+                // skip assemblies that are don't target the correct platform or are non-clr assemblies.
+                complaints = null;
```

Comment:
I'll update the review. 

It seems that BadImageFormatException in the context of ReflectionOnlyLoad only captures scenarios where a native assemblies is inspected in a managed context.  It does not capture scenarios where a managed assembly is targeting x86 and loaded in a 64-bit process - the reflectionOnlyLoad succeeds but the assembly load fails with a bad image exception.

I'll peek at the PE header to see if the assembly is targeting the correct platform.